File: app/src/FizzBuzzType1.php
<?php

namespace App;

class FizzBuzzType1 implements FizzBuzzType
{
    public function generate(int $number): string
    {
        $result = (string) $number;
        if ($number % 3 === 0 && $number % 5 === 0) {
            $result = 'FizzBuzz';
        } elseif ($number % 3 === 0) {
            $result = 'Fizz';
        } elseif ($number % 5 === 0) {
            $result = 'Buzz';
        }

        return $result;
    }
}

File: app/fizz_buzz/lib/types/fizz_buzz_type1.ex
# FizzBuzzタイプ1の実装
defmodule FizzBuzzType1 do
  use FizzBuzzTypeBase

  defimpl FizzBuzzType do
    def generate(_type, number) do
      FizzBuzzTypeBase.basic_fizzbuzz(number, 3, 5, "Fizz", "Buzz")
    end
  end
end